### PR TITLE
Set vmap as the default autograd mode for autodiff cost function.

### PR DESCRIPTION
**Pull request by luisenp**

### CODE REVIEW

**Review by mhmukadam — APPROVED** [review]
Looks good. CI is failing though.

File: theseus/tests/test_theseus_layer.py [hunk 1]
```diff
@@ -125,6 +125,7 @@ def error_fn(optim_vars, aux_vars):
                 th.Vector(cost_weight_dim, name="learnable_err_param", tensor=cw_data)
             ],
             autograd_vectorize=True,
+            autograd_mode="dense",
```
Comment (luisenp) [inline theseus/tests/test_theseus_layer.py:128]: @fantaosha In [this](https://github.com/facebookresearch/theseus/blob/main/theseus/core/cost_function.py#L243) line, `functorch` throws an error when the batch size is not the same for all variables. We could consider automatically broadcasting before passing to `vmap`. Roughly something like the code below (not a clean version)

image

Any concerns about adding something along these lines in a separate PR?

File: theseus/embodied/kinematics/tests/test_inverse_kinematics.py [hunk 2]
```diff
@@ -71,7 +71,12 @@ def ee_pose_err_fn(optim_vars, aux_vars):
     aux_vars = (ee_pose_target,)
 
     cost_function = th.AutoDiffCostFunction(
-        optim_vars, ee_pose_err_fn, 6, aux_vars=aux_vars, name="ee_pose_err_fn"
+        optim_vars,
+        ee_pose_err_fn,
+        6,
+        aux_vars=aux_vars,
+        name="ee_pose_err_fn",
+        autograd_mode="dense",
```
Comment (mhmukadam) [inline theseus/embodied/kinematics/tests/test_inverse_kinematics.py:79]: This does not work with loop_batch or vmap?

Comment (mhmukadam) [reply theseus/embodied/kinematics/tests/test_inverse_kinematics.py:79]: DRM dependency is incompatible.

File: theseus/tests/test_theseus_layer.py (hunk 1 above)
Comment (fantaosha) [reply theseus/tests/test_theseus_layer.py:128]: LGTM.

**Pull request merged**